src/api/java/com/github/minecraftschurlimods/arsmagicalegacy/api/data/SpellPartDataProvider.java

```diff
@@ -31,114 +31,104 @@
 import java.util.function.Supplier;
 
 /**
- * Data provider for spell part data jsons
+ * Base class for spell part data generators.
  */
 public abstract class SpellPartDataProvider implements DataProvider {
+    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
     private static final Logger LOGGER = LogManager.getLogger();
-    private static final Gson GSON = (new GsonBuilder()).setPrettyPrinting().create();
     private final Map<ResourceLocation, JsonObject> data = new HashMap<>();
+    private final String namespace;
     private final DataGenerator generator;
 
-    public SpellPartDataProvider(DataGenerator generator) {
+    public SpellPartDataProvider(String namespace, DataGenerator generator) {
+        this.namespace = namespace;
         this.generator = generator;
     }
 
+    protected abstract void createSpellPartData();
+
     @Override
     public void run(HashCache pCache) {
-        Path path = generator.getOutputFolder();
         createSpellPartData();
-        data.forEach((resourceLocation, jsonObject) -> save(pCache, jsonObject, path.resolve("data/" + resourceLocation.getNamespace() + "/spell_parts/" + resourceLocation.getPath() + ".json")));
+        data.forEach((resourceLocation, jsonObject) -> save(pCache, jsonObject, generator.getOutputFolder().resolve("data/" + resourceLocation.getNamespace() + "/spell_parts/" + resourceLocation.getPath() + ".json")));
     }
 
-    protected abstract void createSpellPartData();
+    @Override
+    public String getName() {
+        return "Spell Part Data";
```

Comment:
The name should be different for each instance with a different output namespace